include/nbl/video/IDeviceMemoryAllocation.h

```diff
@@ -68,32 +70,69 @@ class IDeviceMemoryAllocation : public virtual core::IReferenceCounted
             EMHF_MULTI_INSTANCE_BIT = 0x00000002,
         };
 
+        //! Flags for imported/exported allocation
+        enum E_EXTERNAL_HANDLE_TYPE : uint32_t
+        {
+            EHT_NONE = 0,
+            EHT_OPAQUE_WIN32 = 0x00000002,
+            EHT_OPAQUE_WIN32_KMT = 0x00000004,
+            EHT_D3D11_TEXTURE = 0x00000008,
+            EHT_D3D11_TEXTURE_KMT = 0x00000010,
+            EHT_D3D12_HEAP = 0x00000020,
+            EHT_D3D12_RESOURCE = 0x00000040,
+            EHT_HOST_MAPPED_FOREIGN_MEMORY = 0x00000100,
+        };
+
+        /* ExternalMemoryProperties *//* provided by VK_KHR_external_memory_capabilities */
+        struct SExternalMemoryProperties
+        {
+            uint32_t exportableTypes : 7 = ~0u;
+            uint32_t compatibleTypes : 7 = ~0u;
```

Comment:
btw in c++20, you can use E_EXTERNAL_HANDLE_TYPE  instead of `uint32_t` here   for the bitfield, compiler will pack them just as it would uint32_t because the enum has the same underlying

Reply:
also move the struct to `IPhysicalDevice`